src/notification.c

```diff
@@ -306,31 +318,80 @@ int notification_init(notification * n, int id)
+                                                n->msg,
+                                                &substr,
+                                                n->summary,
+                                                n->markup);
+                                break;
+                        case 'b':
+                                n->msg = notification_replace_single_field(
+                                                n->msg,
+                                                &substr,
+                                                n->body,
+                                                n->markup);
+                                break;
+                        case 'I':
+                                n->msg = notification_replace_single_field(
+                                                n->msg,
+                                                &substr,
+                                                n->icon ? basename(n->icon) : "",
+                                                MARKUP_NO);
+                                break;
+                        case 'i':
+                                n->msg = notification_replace_single_field(
+                                                n->msg,
+                                                &substr,
+                                                n->icon ? n->icon : "",
+                                                MARKUP_NO);
+                                break;
+                        case 'p':
+                                if (n->progress)
+                                        sprintf(pg, "[%3d%%]", n->progress - 1);
+
+                                n->msg = notification_replace_single_field(
+                                                n->msg,
+                                                &substr,
+                                                n->progress ? pg : "",
+                                                MARKUP_NO);
+                                break;
+                        case 'n':
+                                if (n->progress)
+                                        sprintf(pg, "%d", n->progress - 1);
+
+                                n->msg = notification_replace_single_field(
+                                                n->msg,
+                                                &substr,
+                                                n->progress ? pg : "",
+                                                MARKUP_NO);
+                                break;
+                        case '%':
+                                n->msg = notification_replace_single_field(
+                                                n->msg,
+                                                &substr,
+                                                "%",
+                                                MARKUP_NO);
+                                break;
+                        default:
+                                break;
```

Comment:
There should probably be a warning here, let's not silently ignore unknown items.